test/e2e/common/coredns/patch.go

```diff
@@ -0,0 +1,149 @@
+			if err != nil {
+				return err
+			}
+			hosts[fmt.Sprintf("ingress.%s", clusterName)] = gwIP
+			return nil
+		}, retry.Timeout(3*time.Minute), retry.Delay(5*time.Second)); err != nil {
+			return err
+		}
+	}
+	for idx, c := range ctx.Clusters() {
+		clusterName := common.ClusterNames[idx]
+		if err := retry.UntilSuccess(func() error {
+			cm, err := c.Kube().CoreV1().ConfigMaps("kube-system").Get(context.Background(), "coredns", v1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to get coredns config map (cluster=%s): %w", clusterName, err)
+			}
+			if err := updateCorefile(cm, hosts, clusterName); err != nil {
+				return err
+			}
+			_, err = c.Kube().CoreV1().ConfigMaps("kube-system").Update(context.Background(), cm, v1.UpdateOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to update coredns config map (cluster=%s): %w", clusterName, err)
+			}
+			return rolloutRestartDeployment(c, clusterName)
+		}); err != nil {
+			return fmt.Errorf("failed to update configmap coredns (cluster=%s): %w", clusterName, err)
+		}
+	}
+	ctx.Cleanup(func() {
+		for idx, c := range ctx.Clusters() {
+			clusterName := common.ClusterNames[idx]
+			if err := retry.UntilSuccess(func() error {
+				cm, err := c.Kube().CoreV1().ConfigMaps("kube-system").Get(context.Background(), "coredns", v1.GetOptions{})
+				if err != nil {
+					return fmt.Errorf("failed to get coredns config map (cluster=%s): %w", clusterName, err)
+				}
+				cm.Data["Corefile"] = originalCorefilesPerCluster[clusterName]
+				_, err = c.Kube().CoreV1().ConfigMaps("kube-system").Update(context.Background(), cm, v1.UpdateOptions{})
+				if err != nil {
+					return err
+				}
+				return rolloutRestartDeployment(c, clusterName)
+			}); err != nil {
+				scopes.Framework.Errorf("failed to restore configmap coredns (cluster=%s): %v", clusterName, err)
+			}
+		}
+	})
+	return nil
+}
+
+func rolloutRestartDeployment(c cluster.Cluster, clusterName string) error {
+	d, err := c.Kube().AppsV1().Deployments("kube-system").Get(context.Background(), "coredns", v1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get coredns deployment (cluster=%s): %w", clusterName, err)
+	}
+
+	// Modify the Deployment's spec.template.metadata.annotations to trigger a rollout restart
+	if d.Spec.Template.Annotations == nil {
+		d.Spec.Template.Annotations = make(map[string]string)
+	}
+	d.Spec.Template.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
+
+	_, err = c.Kube().AppsV1().Deployments("kube-system").Update(context.TODO(), d, v1.UpdateOptions{})
+	if err != nil {
+		panic(err.Error())
+	}
+	return nil
+}
+
+func updateCorefile(coredns *corev1.ConfigMap, hosts map[string]string, clusterName string) error {
+	const hostsTemplate = `
+    hosts {
+%s
+      fallthrough
+    }`
+
+	var hostsEntries []string
+	for name, ip := range hosts {
+		hostsEntries = append(hostsEntries, fmt.Sprintf("      %s %s", ip, name))
```

Comment:
seeing `      ` boosts my anxiety :) but for this given case it's good enough :D